accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/helper/runtime/RuntimeHelper.java

```diff
@@ -0,0 +1,76 @@
+package biz.netcentric.cq.tools.actool.helper.runtime;
+
+import java.util.Set;
+
+import javax.jcr.Node;
+import javax.jcr.NodeIterator;
+import javax.jcr.Property;
+import javax.jcr.PropertyIterator;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+
+import org.apache.sling.settings.SlingSettingsService;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.startlevel.FrameworkStartLevel;
+
+/** Provides basic context information about the runtime. In own package so it can be inlined with "Conditional-Package" from startup bundle */
+public class RuntimeHelper {
+    
+    private static final String RUNMODE_CLOUD_READY = "cloud-ready";
+
+    public static boolean isCompositeNodeStore(Session session) {
+        
+        try {
+            Node appsNode = session.getNode("/apps");
+            // see https://issues.apache.org/jira/browse/OAK-6563
+            boolean isCompositeNode = !session.hasCapability("addNode", appsNode, new Object[] { "nt:folder" });
+            return isCompositeNode;
+        } catch(Exception e) {
+            throw new IllegalStateException("Could not check if session is connected to a composite node store: "+e, e);
+        }
+    }
+    
+    public static int getCurrentStartLevel() {
+        return getCurrentStartLevel(FrameworkUtil.getBundle(RuntimeHelper.class).getBundleContext());
+    }
+
+    public static int getCurrentStartLevel(BundleContext bundleContext) {
+        return bundleContext.getBundle(Constants.SYSTEM_BUNDLE_ID).adapt(FrameworkStartLevel.class).getStartLevel();
+    }
+    
+    public static boolean isCloudReadyInstance(SlingSettingsService settingsService) {
+        Set<String> runModes = settingsService.getRunModes();
+        boolean isCloudReady = runModes.contains(RUNMODE_CLOUD_READY);
+        return isCloudReady;
+    }
+    
+    public static Node copyNode(Node src, Node dstParent) throws RepositoryException {
```

Comment:
Can't you use https://docs.adobe.com/docs/en/spec/jsr170/javadocs/jcr-2.0/javax/jcr/Workspace.html#copy(java.lang.String,%20java.lang.String)